ArmaForces.Boderator/ArmaForces.Boderator.BotService/Discord/DiscordService.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Discord;
+using Discord.WebSocket;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace ArmaForces.Boderator.BotService.Discord
+{
+    public sealed class DiscordService : IDiscordService, IHostedService
+    {
+        private readonly ILogger<DiscordService> _log;
+        private readonly DiscordSocketClient _discordClient;
+        private string _token;
+
+        public DiscordService(ILogger<DiscordService> logger, string token)
+        {
+            _log = logger;
+            _discordClient = new DiscordSocketClient();
+            _discordClient.Log += message => Task.Run(() => _log.Log(MapSeverity(message.Severity), "[Discord.NET Log] " + message.Message));
+            _token = token;
+        }
+
+        public async Task StartAsync(CancellationToken cancellationToken)
+        {
+            _log.LogInformation("Discord Service started");
+            await _discordClient.LoginAsync(TokenType.Bot, _token, true);
+            await _discordClient.StartAsync();
+            _discordClient.Connected += () => Task.Run(() => _log.LogInformation("Discord connected"));
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken) => Task.Run(() =>
+        {
+            _discordClient.Dispose();
+            _log.LogInformation("Discord Service stopped");
+        }, cancellationToken);
+
+        public string GetDiscordClientStatus()
```

Comment:
Wouldn't it be better to return some `DiscordClientStatus` instead of a string? It would be especially useful for the API.

Reply:
I used it for testing purposes only but I can change this for wider use